### PR TITLE
SW-546_support_multiselect_custom_fields

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: tests/Struct/FindologicProductTest.php [hunk 1]
```diff
@@ -396,6 +396,38 @@ public function testProductWithCustomFields(): void
...
+            $this->getContainer(),
+            $this->salesChannelContext->getContext(),
+            $this->shopkey,
+            $customerGroupEntities,
+            new XMLItem('123')
+        );
+
+        $attributes = $findologicProduct->getCustomFields();
+        foreach ($attributes as $attribute) {
+            $this->assertEquals($attribute->getValues(), $productFields[$attribute->getKey()]);
```
Comment (dblaichinger) [inline tests/Struct/FindologicProductTest.php:427]: expected and actual should be switched.

File: src/Struct/FindologicProduct.php [hunk 2]
```diff
@@ -991,10 +991,19 @@ public function getCustomFields(): array
...
     protected function getCleanedAttributeValue($value)
     {
+        if (is_array($value)) {
+            $values = [];
+            foreach ($value as $item) {
+                $values[] = $this->getCleanedAttributeValue($item);
+            }
+
+            return $values;
+        }
```
Comment (dblaichinger) [inline src/Struct/FindologicProduct.php:1005]: Should handling arrays still be in this method? Or should this logic be put before the call `getCleanedAttributeValue` in `getCustomFieldProperties`?

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Struct/FindologicProduct.php (hunk 2 above)
Comment (TheKeymaster) [reply src/Struct/FindologicProduct.php:1005]: I was thinking the same and I also had both solutions implemented once. I went with this, as this is simple and also allows customers who use the extension plugin to call this method and get their data, cleaned.

File: tests/Struct/FindologicProductTest.php (hunk 1 above)
Comment (TheKeymaster) [reply tests/Struct/FindologicProductTest.php:427]: Oof. Then I might also do that for the other test, from where I copied that ^^.

File: src/Struct/FindologicProduct.php [hunk 3]
```diff
@@ -991,10 +991,19 @@ public function getCustomFields(): array
 
     /**
      * @param string|int|bool $value
```
Comment (dblaichinger) [inline src/Struct/FindologicProduct.php:993]: Add `array<string, int>`

**Review by dblaichinger — CHANGES_REQUESTED** [review]

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**